tools/checkrelease.sh

```diff
@@ -19,114 +19,221 @@
+      make distclean
+      ./tools/configure.sh sim:nsh
+      make
+    fi
+    return_value=$?
+    if [ $return_value -eq 0 ]; then
+      echo " OK: we were able to build sim:nsh"
+    else
+      RETURN_CODE=1
+      echo " - Error building sim:nsh:"
+      echo
+      echo $output
+    fi
+
+    echo
+}
 function usage() {
-    echo "Usage: $0 <URL-of-release-candidate-directory-or-release-name>"
-    echo "   Given release name or release full URL, downloads all files in"
-    echo "   in that directory (which should include nuttx and nuttx-apps"
-    echo "   sha512, asc, and tar.gz files), checks the release SHA512 and GPG "
-    echo "   signatures, and checks the unpacked directories for required "
-    echo "   files. Creates a temporary directory to do its work in."
+    echo "Usage: $0 [--verbose] [--url <URL-of-release-dir>] [--release <name-of-release] [--dir <path-to-directory>] [--tempdir <path-to-directory>]"
+    echo "   Given release full URL, release name, or a local directory, downloads or copies"
+    echo "   all files in that directory (which for a release should include nuttx and nuttx-apps, sha512, "
+    echo "   asc, and tar.gz files), checks the release SHA512 and GPG signatures, checks the unpacked "
+    echo "   directories for required files, and tries to build NuttX for sim:nsh."
+    echo
+    echo "   If tempdir is specified, it will be removed and recreated; if it is not specified, /tmp/nuttx-checkrelease"
+    echo "   is used."
+    echo
+    echo "   If --url or --release are given, nuttx and nuttx-apps tar.gz files are left in /tmp for the user to "
+    echo "   build for their platform."
+    echo
+    echo "   -V and --verbose are equivalent."
     echo
-    echo "   nuttx and nuttx-apps tar.gz files are left in the current"
-    echo "   directory."
-    echo 
     echo "Examples:"
     echo
-    echo "  $0 9.1.0-RC1"
-    echo "  $0 https://dist.apache.org/repos/dist/dev/incubator/nuttx/9.1.0-RC1/"
+    echo "  $0 --release 9.1.0-RC1"
+    echo "  $0 --url https://dist.apache.org/repos/dist/dev/incubator/nuttx/9.1.0-RC1/"
+    echo "  $0 --dir ./some-dir-that-has-nuttx-and-apps"
     echo
 }
 
-if [ "-h" == "$1" ]; then
-    usage
-    exit 0
-fi
+UNKNOWN=()
+while [[ $# -gt 0 ]]
+do
+  key="$1"
 
-if [ -z "$1" ]; then
-    usage
-    exit 0
-fi
+  URL=""
+  DIRECTORY=""
 
-ARG=$1
-if [[ "$ARG" =~ ^"http".* ]]; then
-  URL="$1/"
-else
-  URL="$BASE_URL/$1/"
+  case $key in
+      -U|--url)
+      URL="$2"
+      shift
+      ;;
+      -R|--release)
+      RELEASE="$2"
+      URL="$BASE_URL/$RELEASE/"
+      shift
+      ;;
+      -D|--dir)
+      DIRECTORY="$(readlink -f $2)"
+      shift
+      ;;
+      -T|--tempdir)
+      TEMPDIR="$(readlink -f $2)"
+      shift
+      ;;
+      -h|--help)
+      usage
+      exit 0
+      ;;
+      -V|--verbose)
+      VERBOSE=1
+      ;;
+      *)    # unknown option
+      POSITIONAL+=("$1") # save it in an array for later
+      shift # past argument
```

Comment:
Remove, otherwise we lose the half unknow argument.

Reply:
@xiaoxiang781216 will remove